### DISCUSSION TITLE
Dúvidas de como acessar o retorno do banco no domain.api

### DISCUSSION THREAD

**Discussion by LucasBLs** [balta-io/7196#36] [Perguntas e Respostas] [answered]
Boa tarde,

Tenho duas dúvidas, estou fazendo a autenticação com base em usuários no banco, e gostaria de entender alguns pontos:

- como eu posso verificar se o usuário que veio do banco é null?
No exemplo abaixo eu tenho um command que pega o usuário informado, porém como consigo acessar o retorno do banco na api?
porque nesse caso eu iria retorno que o usuário ou senha é inválido conforme imagem abaixo:
![image](https://user-images.githubusercontent.com/57819294/159129714-8e8b10a0-faeb-43f6-bf8f-c6e7be37dbad.png)

- Seguindo o exemplo acima, como eu posso acessar a propriedade SENHA que retornou do banco para comparar com a que o usuário digitou no diretório da api? ou não deveria ser lá?

Obrigado!

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @LucasBLs, como vai?

Você pode ter o retorno do seu método como um `User?` ao invés de `User`. Isto permite retorno de usuário nulos.
Feito isto basta chegar se ele é nulo com `is null`, `is not null` ou `== null`/``!= null`.

Para acessar a senha, basta navegar para a propriedade... `User.Password` se não me engano.
